websocket/shared.nim

```diff
@@ -183,7 +183,7 @@ proc recvFrame*(ws: AsyncSocket): Future[Frame] {.async.} =
   let opc = b0 and 0x0f
   try:
     f.opcode = opc.Opcode
-  except RangeError:
+  except Exception:
```

Comment:
Can any other exception be raised here, or is this to avoid a `RangeDefect` shim?

Reply:
Come to think of it, `Opcode` is an enum with holes, so the better refactor here would be to check if `opc` is in a valid set of opcodes using something like [`setutils.fullSet`](https://github.com/nim-lang/Nim/blob/55373e65b44c5d15ced134fb59ef24ccfbeee64c/lib/std/setutils.nim#L39-L52). But you don't have to do this here, I can do it myself.